UnitsNet/CustomCode/Quantities/Duration.extra.cs

```diff
@@ -11,13 +11,9 @@ public partial struct Duration
         /// <summary>
         ///     Convert a Duration to a TimeSpan.
         /// </summary>
-        /// <exception cref="ArgumentOutOfRangeException">Throws if the TimeSpan can't represent the Duration exactly </exception>
         /// <returns>The TimeSpan with the same time as the duration</returns>
         public TimeSpan ToTimeSpan()
         {
-            if ( Seconds > TimeSpan.MaxValue.TotalSeconds ||
-                Seconds < TimeSpan.MinValue.TotalSeconds )
-                throw new ArgumentOutOfRangeException( nameof( Duration ), "The duration is too large or small to fit in a TimeSpan" );
```

Comment:
Oh, I guess this was here for a reason. It seems TimeSpan simply rolls over to negative if exceeding the max value. I don't like that, let's keep our checks in place and throw here.

Reply:
Added back